packages/truffle-core/lib/debug/object-compiler.js

```diff
@@ -0,0 +1,73 @@
+const compile = require("truffle-compile");
+const fs = require("fs");
+
+// this class copes with a directory of .sol files each if which
+// becomes a property of an object which is then compiled
+// the imports in these files will be in the form
+// import "file"
+// rather than import "file.sol"
+class DebugObjectCompiler {
+  constructor(config) {
+    this.config = config;
+  }
+  async compile() {
+    // all the file names
+    const fileNames = await new Promise((resolve, reject) =>
+      fs.readdir(this.config.contracts_directory, function(err, filePaths) {
+        if (err) {
+          return reject(err);
+        }
+        return resolve(filePaths);
+      })
+    );
```

Comment:
I think you can replace this with [truffle-contract-sources](https://github.com/trufflesuite/truffle/blob/develop/packages/truffle-contract-sources/index.js). You can see how it's used [here](https://github.com/trufflesuite/truffle/blob/a1f1340c525d8a1e667386ffd8c61d948ad5d315/packages/truffle-compile/index.js#L355) for instance.

Reply:
I also changed it to use promisify that regex is a bit ugly but it works + for both / and \